packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2051,11 +2051,19 @@ def _schedule_round(self, round_cls: AppState) -> None:
 
         # self.state will point to last result,
         # or if not available to the initial state
-        last_result = (
-            self._round_results[-1]
-            if len(self._round_results) > 0
-            else self._initial_synchronized_data
-        )
+        if self._round_results:
+            synchronized_data_class = getattr(
+                round_cls, "synchronized_data_class", None
+            )
+            if not synchronized_data_class:
+                self.logger.warning(f"No `synchronized_data_class` set on {self}")
+                last_result = self._round_results[-1]
+            else:
+                synchronized_data_class = round_cls.synchronized_data_class  # type: ignore
+                last_result = synchronized_data_class(db=self._round_results[-1].db)
+                self._round_results[-1] = last_result
+        else:
+            last_result = self._initial_synchronized_data
```

Comment:
we can only cast if the `synchronized_data_class` is set, in order for this to not constitute a breaking change. This has been added as an item to the refactoring lists for open-autonomy.

Reply:
to clarify: this is done here and not during the instantiation of the next `round_cls` in order for the `AbciApp.synchronized_data` to behave in accordance
https://github.com/valory-xyz/open-autonomy/blob/05fb30f861114c2f1fb9b8731c748356596b7cb1/packages/valory/skills/abstract_round_abci/base.py#L1920-L1928